File: app/src/main/java/com/asfoundation/wallet/ui/iab/FiatValue.java
package com.asfoundation.wallet.ui.iab;

import java.io.Serializable;

/**
 * Created by franciscocalado on 26/07/2018.
 */

public class FiatValue implements Serializable {
  private double amount;
  private String currency;

  public FiatValue(double amount, String currency) {
    this.amount = amount;
    this.currency = currency;
  }

  public double getAmount() {
    return amount;
  }

  public String getCurrency() {
    return currency;
  }
}

File: app/src/main/java/com/asfoundation/wallet/ui/iab/FiatValue.kt
package com.asfoundation.wallet.ui.iab

import java.io.Serializable
import java.math.BigDecimal

data class FiatValue(val amount: BigDecimal, val currency: String, val symbol: String = "") :
    Serializable {

  override fun equals(other: Any?) = other is FiatValue
        && other.amount.compareTo(this.amount) == 0
        && other.currency == this.currency
        && other.symbol == this.symbol
}
